app/support/products/scheduling_support.rb

```diff
@@ -152,39 +152,49 @@ def update_schedule_name
 
   class ReservationFinder
 
-    attr_accessor :time, :rule, :day_start, :day_end, :options
+    attr_reader :rule, :day_start, :day_end, :options, :start_time
 
     def initialize(time, rule, options = {})
-      self.time = rule.instrument.cutoff_hours > 0 ? [time, rule.instrument.cutoff_hours.hours.from_now].max : time
-      self.rule = rule
-      self.day_start = Time.zone.local(time.year, time.month, time.day, rule.start_hour, rule.start_min, 0)
-      self.day_end   = Time.zone.local(time.year, time.month, time.day, rule.end_hour, rule.end_min, 0)
-      self.options   = options
-      adjust_time
-    end
-
-    def adjust_time
-      # we can't start before the rules say we can
-      self.time = day_start if time < day_start
-
-      # check for conflicts with rule interval/duration time and adjust to next interval if necessary
-      duration_mins = rule.instrument.reserve_interval.to_i
-      self.time += (duration_mins - time.min % duration_mins).minutes unless time.min % duration_mins == 0
+      @options    = options
+      @rule       = rule
+      @day_start  = Time.zone.local(time.year, time.month, time.day, rule.start_hour, rule.start_min, 0)
+      @day_end    = Time.zone.local(time.year, time.month, time.day, rule.end_hour, rule.end_min, 0)
+      @start_time = set_start_time(time)
     end
 
     def next_reservation(reserver, duration)
-      start_time = time
+      next_time = start_time
 
-      while start_time < day_end
-        reservation = reserver.reservations.new(reserve_start_at: start_time, reserve_end_at: start_time + duration)
+      while next_time < day_end
+        reservation = reserver.reservations.new(reserve_start_at: next_time, reserve_end_at: next_time + duration)
 
         conflict = reservation.conflicting_reservation(exclude: options[:exclude])
         return reservation if conflict.nil?
 
-        start_time = conflict.reserve_end_at
+        next_time = conflict.reserve_end_at
       end
     end
 
+    private
+
+    def set_start_time(time)
+      if !options[:ignore_cutoff] && rule.instrument.cutoff_hours > 0
```

Comment:
Not a big deal, but this might be easier to read as `if options[:ignore_cutoff] || rule.instrument.cutoff_hours <= 0` and flip the conditional (I think I got that right). Not a fan of double negatives (not & ignore in this case).